web/downloadService/get_private_filter_download_test.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+	if err := sendV4FileToAWS(region, bucketName, fileName); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+	if err := mongo.Setup(filterDoc, filterBlueprintDoc); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+	downloadService := httpexpect.New(t, cfg.DownloadServiceURL)
+
+	Convey("Given a published version exists with a private link", t, func() {
+		Convey("When a request is made for the private document", func() {
+			Convey("Then the response streams the decrypted private file", func() {
+
+				response := downloadService.GET("/downloads/filter-outputs/{filterOutputID}.csv", filterOutputID).
+					Expect().Status(http.StatusOK)
+
+				response.Body().Equal(string(f))
+			})
+		})
+	})
+
+	if err := mongo.Teardown(filterDoc, filterBlueprintDoc); err != nil {
+		if err != mgo.ErrNotFound {
+			log.ErrorC("Failed to tear down test data", err, nil)
+			os.Exit(1)
+		}
+	}
+
+	if err := deleteS3File(region, bucketName, fileName); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+}
+
+func TestPrivateFilterDownloadDecryptedAndStreamedUnpublishedWithoutAuthentication(t *testing.T) {
+	if len(os.Getenv("VAULT_ADDR")) == 0 || len(os.Getenv("VAULT_TOKEN")) == 0 {
+		log.Info("failing test as no vault token or address set - use make test", nil)
+		t.FailNow()
+	}
+
+	filterID := uuid.NewV4().String()
+	filterBlueprintID := uuid.NewV4().String()
+	filterOutputID := uuid.NewV4().String()
+	datasetID := uuid.NewV4().String()
+	instanceID := uuid.NewV4().String()
+	edition := "2017"
+	version := 1
+
+	isPublished := false
+
+	filterBlueprintDoc := &mongo.Doc{
+		Database:   cfg.MongoFiltersDB,
+		Collection: "filters",
+		Key:        "_id",
+		Value:      filterID,
+		Update:     filterAPI.GetValidFilterWithMultipleDimensionsBSON(cfg.FilterAPIURL, filterID, instanceID, datasetID, edition, filterBlueprintID, version, isPublished),
+	}
+
+	filterDoc := &mongo.Doc{
+		Database:   cfg.MongoFiltersDB,
+		Collection: "filterOutputs",
+		Key:        "_id",
+		Value:      filterID,
+		Update:     filterAPI.GetValidFilterOutputWithPrivateDownloads(cfg.FilterAPIURL, filterID, instanceID, filterOutputID, filterBlueprintID, datasetID, edition, version, isPublished),
+	}
+
+	if err := sendV4FileToAWS(region, bucketName, fileName); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+	if err := mongo.Setup(filterDoc, filterBlueprintDoc); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+	downloadService := httpexpect.New(t, cfg.DownloadServiceURL)
+
+	Convey("Given an associated version exists with a private link", t, func() {
+		Convey("When a request is made for the private document without authentication", func() {
+			Convey("Then the response returns a not found http status", func() {
```

Comment:
and yet the below is a server error...

Reply:
Fixed description and added comment - it should really be a 404 once we update to use go-ns clients